packages/app/components/jsonschemaform/jsonschemaform.tsx

```diff
@@ -63,6 +64,14 @@ const JsonSchemaForm = ({
 
     const { _id, ...document } = formData
 
+    //* This function gets called with formEl.current.validate().
+    //* You can create any custom validation needed and add it to this stack.
+    function validate(formData: any, errors: FormValidation) {
```

Comment:
This wrapper function is in this lexical scope because it's very likely that validation will need access `formEl` for some property or internal state. Future custom validations could become part of the call stack here.